tests/unit/base_tests_zarrio.py

```diff
@@ -502,60 +567,79 @@ def test_write_attributes_write_3Darray_of_floats(self):
     def test_write_attributes_write_reference_to_datasetbuilder(self):
         data_1 = np.arange(100, 200, 10).reshape(2, 5)
         dataset_1 = DatasetBuilder('dataset_1', data_1)
-        testgroup = self.io._ZarrIO__file  # For testing we just use our file and create some attributes
+        tempIO = ZarrIO(self.store, mode='w')
+        tempIO.open()
+        testgroup = tempIO.file  # For testing we just use our file and create some attributes
         attr = {'attr1': dataset_1}
-        self.io.write_attributes(testgroup, attr)
+        tempIO = ZarrIO(self.store, mode='w')
+        tempIO.write_attributes(testgroup, attr)
         expected_value = {'attr1': {'zarr_dtype': 'object', 'value': {'source': ".", 'path': '/dataset_1'}}}
         self.assertDictEqual(testgroup.attrs.asdict(), expected_value)
+        tempIO.close()
```

Comment:
why does this need to be closed but not the above methods like `test_write_attributes_write_scalar_str` which also call
```
tempIO = ZarrIO(self.store, mode='w')
tempIO.open()
tempIO.write_attributes(...)
```

Reply:
In principle it is no longer necessary (as of the latest HDMF release) to call close here explicitly, because this is now taken care of by ``HDMFIO.__del__`` when the object is being garbage collected at the end of the function. However, in the test cases where we need to call ``open`` explicitly to test the low-level write functions, I think it is cleaner to then also explicitly call ``close``. I updated the tests to add missing ``close`` calls. Note, this is only for tests that use the low level write functions directly (e.g., to test ``write_dataset``). Most other tests can use the more standard pattern of using a context for the IO object so that we don't have to explicitly do the open and close.